waves/src/App.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from "react";
+import "./App.css";
+
+function App() {
+    const [welcome, setWelcome] = useState<String>("Not welcome yet");
+    const [rate, setRate] = useState<String>("No rate received yet");
+
+    import("./wallet-lib/pkg").then(({ hello }) => {
+        let welcome = hello("World");
+        setWelcome(welcome);
+    });
```

Comment:
I feel this is not gonna be the endgame of how we can/want to use wasm functionality.

What happens if you call this code twice? Will it load the wasm file twice from the server?
Another interesting thing to explore would be to hide this behind a regular `async` function. `import()` simply returns a `Promise` which means you can await it. As such, you could just expose a `async function hello()` in TS and we wouldn't need to be concerned here with the fact that we need to import the wasm dynamically. That ofc only makes sense if we don't load the WASM again for every function call but this is somehow cached by webpack!

Reply:
If it is not cached, we might want to abstract this away in a custom hook that caches the wasm instance using `useState`, similar to what [this](https://codeburst.io/getting-started-with-react-and-webassembly-using-hooks-441818c91608) blogpost is doing!

Happy to look into this tomorrow together if you want :)

The reason I am pushing for this is because I think it would be good if we have a relatively clear idea of how we are going to consume the WASM code from within React that doesn't cause a lot of boilerplate!

Reply:
I agree, this is not optimal. Happy to look into this together :)